src/app/Reducers/userReducer.js

```diff
@@ -7,25 +7,25 @@ export default function userReducer(state=initialState, action) {
 
   switch (action.type) {
     case ADD_FRIEND:
-      return [{
-        id
-      }, ...state];
+      return Object.assign({}, state, {
+        friends: state.friends.concat([action.friend])
+      });
 
     case REMOVE_FRIEND:
       var friends = state.friends.filter(friend => {
-        return friend.id !== action.id
+        return friend.id !== action.id;
       });
-      return Object.assign({}, { profile: state.profile, friends: friends });
+      return Object.assign({}, state, { friends: friends });
```

Comment:
update the part of the state that we want to change then pass all properties to a new object so we have a new object reference for the state